### PR TITLE
MRG: Picks argument to epochs.get_data 

### LINKED ISSUE #6171: picks argument to get_data
Currently, if you want to do e.g. decoding, you'll do something like `X = epochs.get_data()`. Now this will often pull in the stim channel, which might e.g. lead to perfect accuracy at t=0. So you would do: `X = epochs.pick_types(stim=False).get_data()` or something like that.

This isn't too inelegant code wise, but 1. it's easy to forget, 2. it's wasteful - we construct a new Epochs object just to remove one chan! It would be more efficient if get_data could take some argument to indicate what types to pick. Then, we could simply index into the returned array, instead of constructing a completely new epochs object. E.g.:

`X = epochs.get_data(stim=False)`

API wise, we could:
- accept the same arguments as pick_types
- accept a dict of types to bools, or a collection of strings
- take a picks argument

Thoughts?

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Closes #6171

What about an exclude param (for exclude="bads")?

### CODE REVIEW

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -166,7 +167,7 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
     if combine is not None:
         ts_args["show_sensors"] = False
 
-    if picks is None:
+    if picks is None or not _is_numeric(picks):
```
Comment (jona-sassenhagen) [inline mne/viz/epochs.py]: This is so that you don't get 300 epochs images if you do `epochs.plot_image(picks=["grad", "mag"])`

**Comment by larsoner** [comment]
> What about an exclude param (for exclude="bads")?

I would say no, if people need this level of specificity they should use `pick_types` directly

**Comment by jona-sassenhagen** [comment]
Wait a sec, I will update examples.

**Comment by jona-sassenhagen** [comment]
Also it's not documented :D

File: mne/epochs.py [hunk 2]
```diff
@@ -1220,17 +1224,26 @@ def _get_data(self, out=True, verbose=None):
...
 
-    def get_data(self):
+    def get_data(self, picks=None):
         """Get all epochs as a 3D array.
 
         Returns
         -------
         data : array of shape (n_epochs, n_channels, n_times)
             A view on epochs data.
+        %(picks_all)s
```
Comment (jona-sassenhagen) [inline mne/epochs.py]: @larsoner is this how the docstring should be?

Comment (larsoner) [reply mne/epochs.py]: It either needs `@verbose` decorator (and arg, so not relevant here) or `@fill_doc` decorator, otherwise it looks good

**Comment by jona-sassenhagen** [comment]
I had to do something on RtEpochs. I don't know that part of the code. Maybe @teonbrooks @jasmainak do? Can you check?

**Comment by larsoner** [comment]
Please set title to MRG if it's good to go from your end @jona-sassenhagen

File: examples/preprocessing/plot_eog_artifact_histogram.py [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
 # Get the stim channel data
 pick_ch = mne.pick_channels(epochs.ch_names, ['STI 014'])[0]
-data = epochs.get_data()[:, pick_ch, :].astype(int)
+data = epochs.get_data(picks=pick_ch).astype(int)
```
Comment (larsoner) [inline examples/preprocessing/plot_eog_artifact_histogram.py]: dimensionality change here [broke circleci](https://circleci.com/gh/mne-tools/mne-python/12491)

Comment (larsoner) [reply examples/preprocessing/plot_eog_artifact_histogram.py]: (it has a singleton dimension where it did not before)

**Comment by jona-sassenhagen** [comment]
@drammock I have moved epochs objects out of most of the helper functions and made everything a bit more modular. This should now be more conductive to looping over axes and data and calling functions to handle and eventually plot arrays.

**Comment by jona-sassenhagen** [comment]
@larsoner I don't understand this error:

```
        if len(incorrect) > 0:
>           raise AssertionError(msg)
E           AssertionError: 
E           mne.epochs.BaseEpochs.get_data arg mismatch: ['picks']
E           mne.epochs.Epochs.get_data arg mismatch: ['picks']
E           mne.epochs.EpochsArray.get_data arg mismatch: ['picks']
E           mne.epochs.RtEpochs.get_data arg mismatch: ['picks']
```

I added the parameter to the methods in BaseEpochs and RtEpochs. If I understand it correctly, the others inherit theirs?

**Comment by GuillaumeFavelier** [comment]
Also that's the first time I see this traceback after all the tests I did myself, could you maybe restart the pipeline again? We could ping the `vtki` team about it in a second step to see if they have an idea.
```
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-req-build-8wjtddob\setup.py", line 474, in <module>
      **config
    File "D:\a\1\s\conda\envs\test\lib\site-packages\numpy\distutils\core.py", line 171, in setup
      return old_setup(**new_attr)
    File "D:\a\1\s\conda\envs\test\lib\site-packages\setuptools\__init__.py", line 145, in setup
      return distutils.core.setup(**attrs)
    File "D:\a\1\s\conda\envs\test\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "D:\a\1\s\conda\envs\test\lib\distutils\dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "D:\a\1\s\conda\envs\test\lib\distutils\dist.py", line 974, in run_command
      cmd_obj.run()
    File "D:\a\1\s\conda\envs\test\lib\site-packages\wheel\bdist_wheel.py", line 192, in run
      self.run_command('build')
    File "D:\a\1\s\conda\envs\test\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "D:\a\1\s\conda\envs\test\lib\distutils\dist.py", line 974, in run_command
      cmd_obj.run()
    File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-req-build-8wjtddob\setup.py", line 268, in run
      build_tvtk_classes_zip()
    File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-req-build-8wjtddob\setup.py", line 254, in build_tvtk_classes_zip
      gen_tvtk_classes_zip()
    File "tvtk\setup.py", line 83, in gen_tvtk_classes_zip
      from tvtk.code_gen import TVTKGenerator
    File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-req-build-8wjtddob\tvtk\code_gen.py", line 10, in <module>
      import vtk_module as vtk
    File "tvtk\vtk_module.py", line 15, in <module>
      from vtk import *
  ModuleNotFoundError: No module named 'vtk'
```

**Comment by larsoner** [comment]
@GuillaumeFavelier it looks like for some reason it's trying to build Mayavi (!), since that is what creates tvtk. And it has a problem importing `vtk`, can you do `import vtk` in your terminal?

**Comment by jona-sassenhagen** [comment]
Btw I think I have restarted Azure, see https://github.com/mne-tools/mne-python/pull/6198/checks?check_run_id=109638796

File: mne/epochs.py [hunk 4]
```diff
@@ -1220,17 +1225,27 @@ def _get_data(self, out=True, verbose=None):
...
-    def get_data(self):
+    @fill_doc
+    def get_data(self, picks=None):
         """Get all epochs as a 3D array.
 
         Returns
         -------
         data : array of shape (n_epochs, n_channels, n_times)
             A view on epochs data.
+        %(picks_all)s
```
Comment (larsoner) [inline mne/epochs.py]: This is in the `Returns` section ...

**Comment by GuillaumeFavelier** [comment]
Thank you @jona-sassenhagen and apparently it is the same result. It seems to happen on windows in the conda only build. I dont reproduce it on my linux @larsoner, I'll try on my windows. I could open an issue and patch it for now, what do you think?

**Comment by larsoner** [comment]
It's probably this

https://github.com/conda/conda/issues/8187

**Comment by larsoner** [comment]
Actually that error looks non-fatal. Later it does install fine. I'll look into Azure a bit and see if I can fix it

File: mne/epochs.py (hunk 4 above)
Comment (jona-sassenhagen) [reply mne/epochs.py]: giphy

Comment (jona-sassenhagen) [reply mne/epochs.py]: And with that, `make docstrings` works for me locally.

**Comment by larsoner** [comment]
Thanks @jona-sassenhagen

**Pull request merged**